crates/ruff_linter/src/checkers/ast/mod.rs

```diff
@@ -1794,7 +1794,6 @@ impl<'a> Checker<'a> {
         self.visit_expr(&generator.iter);
         self.semantic.push_scope(ScopeKind::Generator(kind));
 
-        self.semantic.flags = flags | SemanticModelFlags::COMPREHENSION_ASSIGNMENT;
```

Comment:
Good to remove, looks like we got rid of this `BindingKind` at some point: https://github.com/astral-sh/ruff/pull/9967